Protos/V1/sessions_filters.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+
+package armonik.api.grpc.v1.sessions;
+
+import "filters_common.proto";
+import "google/protobuf/timestamp.proto";
+import "session_status.proto";
+import "sessions_fields.proto";
+
+option csharp_namespace = "ArmoniK.Api.gRPC.V1.Sessions";
+
+message FilterString {
+  SessionField field = 1;
+  string value = 2;
+  FilterStringOperator operator = 3;
+}
+
+message FilterNumber {
+  SessionField field = 1;
+  int64 value = 2;
+  FilterNumberOperator operator = 3;
+}
+
+message FilterDate {
+  SessionField field = 1;
+  google.protobuf.Timestamp value = 2;
+  FilterDateOperator operator = 3;
+}
+
+message FilterBoolean {
+  SessionField field = 1;
+  bool value = 2;
+}
+
+message FilterArray {
+  SessionField field = 1;
+  string value = 2;
+  FilterArrayOperator operator = 3;
+}
+
+message FilterStatus {
+  SessionField field = 1;
+  session_status.SessionStatus value = 2;
+  FilterStatusOperator operator = 3;
+}
+
+message FilterField {
+  oneof filter {
+    FilterString string = 1;
+    FilterNumber number = 2;
+    FilterDate date = 3;
+    FilterBoolean boolean = 4;
+    FilterArray array = 5;
+    FilterStatus status = 6;
+  }
+}
```

Comment:
Sorry I'm a bit late to the party, but I think the following could be better.

Like that, you could reuse `FilterString`, `FilterNumber`, `FilterValue`... across all services.

```suggestion

message FilterString {
  string value = 1;
  FilterStringOperator operator = 2;
}

message FilterNumber {
  int64 value = 1;
  FilterNumberOperator operator = 2;
}

message FilterValue {
  oneof condition {
    FilterString string = 1;
    FilterNumber number = 2;
    // ...
  }
}

message FilterField {
  SessionField field = 1;
  FilterValue value_condition;
}
```

Reply:
I don't know how much we'll be able to factorize on my end but let's try !

Reply:
I update proto to support this suggestion. thanks @lemaitre-aneo